core/include/scipp/core/variable.tcc

```diff
@@ -613,6 +614,52 @@ Variable::Variable(const units::Unit unit, const Dimensions &dimensions,
                                                     std::move(values_),
                                                     std::move(variances_))) {}
 
+template <class T, class... T1, class... T2>
```

Comment:
Some of the added code shouldn't be in `.tcc` file, but there are some dependencies on constructors defined here. All the stuff is going to be changed after substituting `makeVariable` to this constructor, so it would be fixed.